titus-server-master/src/main/java/com/netflix/titus/master/clusteroperations/ClusterAgentAutoScaler.java

```diff
@@ -360,13 +368,15 @@ Completable doAgentScaling() {
                                                         Map<String, Long> numberOfTasksOnAgent,
                                                         long finished,
                                                         long elapsed) {
+        // use reverse order of instance group state such that PhasedOut instances are scaled down first.
         return instancesForActiveInstanceGroups.entrySet().stream()
                 .filter(e -> {
                     AgentInstanceGroup instanceGroup = e.getKey();
                     return instanceGroup.getTier() == tier &&
                             instanceGroup.getInstanceType().equals(primaryInstanceType) &&
                             !instanceGroup.getAttributes().containsKey(ClusterOperationsAttributes.NOT_REMOVABLE);
                 })
+                .sorted(IDLE_INSTANCE_COMPARATOR)
```

Comment:
if my read of `createSetRemovableOverrideStatusesCompletable()` is correct, it doesn't seem this order will influence which instance groups get picked to be scaled down first. The order for scaling down seems to be also controlled by `getActiveInstanceGroups()`.

Reply:
more specifically, here: https://github.com/corindwyer/titus-control-plane/blob/scale-phased-out-agents/titus-server-master/src/main/java/com/netflix/titus/master/clusteroperations/ClusterAgentAutoScaler.java#L434

which gets called with the list of instance groups being `activeInstanceGroupsForTier`, here: https://github.com/corindwyer/titus-control-plane/blob/scale-phased-out-agents/titus-server-master/src/main/java/com/netflix/titus/master/clusteroperations/ClusterAgentAutoScaler.java#L302